istioctl/cmd/istioctl/meshdump.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"istio.io/istio/istioctl/pkg/kubernetes"
+	"istio.io/istio/pkg/log"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+)
+
+var (
+	outMeshDumpFilename string
+
+	meshDumpCmd = &cobra.Command{
+		Use:   "mesh-dump",
+		Short: "Creates an archive of mesh information for debugging purposes [kube only]",
```

Comment:
Is the intention for this command to also capture _all_ mesh information, e.g. mixer, galley, citadel? If not, could we rename this commands to be more specific to Pilot.

It might also be good to move these under experimental until the output format is stable.

Reply:
Agreed on the experimental point.

The idea is for a user to run this command to give one of us as much information as possible to debug with so will eventually include citadel, mixer, galley, etc. information. This is just a first pass.